# Fix pkg_tar to work with a in-build python interpreter.
# See https://github.com/bazelbuild/rules_pkg/issues/926.
--- pkg/private/tar/tar.bzl
+++ pkg/private/tar/tar.bzl
@@ -187,16 +187,21 @@
+    py3_runtime = ctx.toolchains["@bazel_tools//tools/python:toolchain_type"].py3_runtime
     ctx.actions.run(
         mnemonic = "PackageTar",
         progress_message = "Writing: %s" % output_file.path,
         inputs = inputs,
-        tools = [ctx.executable.compressor] if ctx.executable.compressor else [],
+        tools = (
+            [ctx.executable.compressor] if ctx.executable.compressor else [] +
+            [py3_runtime.files] if py3_runtime.files else []
+        ),
         executable = ctx.executable._build_tar,
         arguments = [args],
         outputs = [output_file],
         env = {
             "LANG": "en_US.UTF-8",
             "LC_CTYPE": "UTF-8",
             "PYTHONIOENCODING": "UTF-8",
             "PYTHONUTF8": "1",
+            "PATH": "$PATH" + (":" + py3_runtime.interpreter.dirname if py3_runtime.interpreter else ""),
         },
         use_default_shell_env = True,
     )
@@ -310,3 +315,4 @@
         ),
     },
+    toolchains = ["@bazel_tools//tools/python:toolchain_type"],
 )